implementations/utils/metric.py

```diff
@@ -397,8 +396,38 @@ def is_valid_entry() -> bool:
   return True
 
 
-def get_job_status(benchmark_id: str) -> bigquery.JobStatus:
-  """Get job status for the run.
+def get_gke_job_status(benchmark_id: str) -> bigquery.JobStatus:
+  """Get job status for the GKE run.
+
+  FAILED - if any failure occurs in run_model
+  SUCCESS - end-to-end model tests are successful in run_model
+  """
+  context = get_current_context()
+  execution_date = context["dag_run"].logical_date
+  current_dag = context["dag"]
+
+  workload_completion = current_dag.get_task(
```

Comment:
This DAG introspection looks kind of sketchy to me. Could the status here be plumbed through XCOM instead?

Reply:
I did not find a good way to get the status when running `run_workload` via bash script. There is a bug for xpk to wait till completion (b/306688270). I tried command in that bug to get status of workload, but unfortunately, we will need extra efforts to clean up output (output including extra logs & tables).

Reply:
One example for status: https://screenshot.googleplex.com/BpDxsaUX6nh6BJV We will need to parse both status and status message to see if this job is failed or successful.

Reply:
Got it. Let's revisit this if/when XPK adds a Python API and see if there's a better way to plumb the status.